astropy/stats/bayesian_blocks.py

```diff
@@ -45,19 +45,31 @@
    https://www.tandfonline.com/doi/abs/10.1080/01621459.1969.10501038
 """
 
+from __future__ import annotations
+
 import warnings
 from inspect import signature
+from typing import TYPE_CHECKING
 
 import numpy as np
 
 from astropy.utils.exceptions import AstropyUserWarning
 
+if TYPE_CHECKING:
+    from numpy.typing import ArrayLike, NDArray
+
 # TODO: implement other fitness functions from appendix C of Scargle 2013
 
 __all__ = ["FitnessFunc", "Events", "RegularEvents", "PointMeasures", "bayesian_blocks"]
 
 
-def bayesian_blocks(t, x=None, sigma=None, fitness="events", **kwargs):
+def bayesian_blocks(
+    t: ArrayLike,
+    x: ArrayLike | None = None,
+    sigma: ArrayLike | float | None = None,
+    fitness: str | FitnessFunc = "events",
```

Comment:
With `from typing import Literal`, I think this could be:

```suggestion
    fitness: Literal["events", "regular_events", "measures"] | FitnessFunc = "events",
```